### ISSUE TITLE
Delimit weapon selection

### ISSUE THREAD

**Issue by GreenLunar** [assaultcube/AC#122]
There are players who like to play OSOK with knife only, and they ask other players to use only knife; such is a round with players who are voluntarily agree to use a specific weapon, but, sometimes, not all would respect that request.

Delimiting weapon selection by hosting server would allow to enforce more challenging games, be it SURV, OSOK, LSS, DM, (C/H/K)TF and even PF (by restricting grenades and knife).

Such mechanism, activated even when no weapons are restricted, might even harden cheaters from using grenades on OSOK games.

Example:
`/osok ac_douze 10 k` (s stands for sniper)`
`/dm ac_douze 10 s` (s stands for sniper)`

**Comment by ac-stef** [comment]
We generally don't allow servers to modify gameplay - at least not when they are listed on the masterserver.

Knife-only games can be played quite simple: play LSS on a map without grenade pickups. The fact, that there are almost no modded maps with no pickups around, tells me, that there's not that much demand for knife-only games.

**Comment by GreenLunar** [comment]
> Knife-only games can be played quite simple: play LSS on a map without grenade pickups.

This is not true.  Even if so, there is no way to turn initial players health points to a 1%.

**Comment by ac-stef** [comment]
Of course it is true.

**Comment by GreenLunar** [comment]
Please explain.

**Comment by ac-stef** [comment]
Playing LSS on a map without nade pickups results in a knife-only game.

**Comment by GreenLunar** [comment]
Yes, but I am limited to specific maps, and not all servers allow to have extra maps (i.e. making only official maps available).

Also, currently, it is not possible to play LSS with 1% of health points, unless all players voluntarily switch to knife.
